community/security/src/test/java/org/neo4j/server/security/auth/RateLimitedAuthenticationStrategyTest.java

```diff
@@ -120,6 +122,31 @@ public void shouldSlowRequestRateOnMultipleFailedAttemptsWhereAttemptIsValid() t
         assertThat( authStrategy.authenticate( user, "right" ), equalTo( AuthenticationResult.SUCCESS ) );
     }
 
+    @Test
+    public void shouldAllowUnlimitedFailedAttemptsWhenMaxFailedAttemptsIsZero()
+    {
+        testUnlimitedFailedAuthAttempts( 0 );
+    }
+
+    @Test
+    public void shouldAllowUnlimitedFailedAttemptsWhenMaxFailedAttemptsIsNegative()
+    {
+        testUnlimitedFailedAuthAttempts( -42 );
+    }
+
+    private void testUnlimitedFailedAuthAttempts( int maxFailedAttempts )
+    {
+        FakeClock clock = getFakeClock();
+        AuthenticationStrategy authStrategy = new RateLimitedAuthenticationStrategy( clock, maxFailedAttempts );
+        User user = new User.Builder( "user", Credential.forPassword( "right" ) ).build();
+
+        int attempts = ThreadLocalRandom.current().nextInt( 5, 100 );
+        for ( int i = 0; i < attempts; i++ )
+        {
+            assertEquals( AuthenticationResult.TOO_MANY_ATTEMPTS, authStrategy.authenticate( user, "wrong" ) );
```

Comment:
I don't understand the idea behind this test/name. The unlimited part seems to me to indicate that we never should get `TOO_MANY_ATTMPTS` but the test is written to assert that it always says that from the first try...